src/main/java/com/crowdin/client/Crowdin.java

```diff
@@ -162,6 +198,55 @@ public File downloadTranslations(VirtualFile sourceFile, String branch) {
         }
     }
 
+    public Map<Long, Directory> getDirectories(Long branchId) {
+        try {
+            return executeRequestFullList((limit, offset) ->
+                    this.client.getSourceFilesApi()
+                        .listDirectories(this.projectId, branchId, null, null, limit, offset)
+                        .getData()
+                )
+                .stream()
+                .map(ResponseObject::getData)
+                .filter(dir -> Objects.equals(dir.getBranchId(), branchId))
+                .collect(Collectors.toMap(Directory::getId, Function.identity()));
+        } catch (Exception e) {
+            throw new RuntimeException(this.getErrorMessage(e));
+        }
+    }
+
+    public List<com.crowdin.client.sourcefiles.model.File> getFiles(Long branchId) {
+        try {
+            return executeRequestFullList((limit, offset) ->
+                    this.client.getSourceFilesApi()
+                        .listFiles(this.projectId, branchId, null, null, 500, 0)
+                        .getData()
+                )
+                .stream()
+                .map(ResponseObject::getData)
+                .filter(file -> Objects.equals(file.getBranchId(), branchId))
+                .collect(Collectors.toList());
+        } catch (Exception e) {
+            throw new RuntimeException(this.getErrorMessage(e));
+        }
+    }
+
+    /**
+     * @param request represents function that downloads list of models and has two args (limit, offset)
+     * @param <T> represents model
+     * @return list of models accumulated from request function
+     */
+    private static <T> List<T> executeRequestFullList(BiFunction<Integer, Integer, List<T>> request) {
```

Comment:
same comment as for placeholder util. Nice candidate to move it to the client. But not now)